integration_test/cleanup_ignore_demo_test.dart

```diff
@@ -0,0 +1,282 @@
+
+      final demoButton = find.text('Demo');
+      expect(demoButton, findsOneWidget);
+
+      await tester.tap(demoButton);
+      await tester.pumpAndSettle();
+
+      await tester.pump(pause);
+
+      final dsPathTextFinder = find.byKey(datasetPathKey);
+      expect(dsPathTextFinder, findsOneWidget);
+      final dsPathText = dsPathTextFinder.evaluate().first.widget as TextField;
+      String filename = dsPathText.controller?.text ?? '';
+      expect(filename, 'rattle::weather');
+
+      // Find the right arrow button in the PageIndicator.
+      final rightArrowFinder = find.byIcon(Icons.arrow_right_rounded);
+      expect(rightArrowFinder, findsOneWidget);
+
+      // Tap the right arrow button to go to the variable role selection.
+      await tester.tap(rightArrowFinder);
+      await tester.pumpAndSettle();
+      await tester.tap(rightArrowFinder);
+      await tester.pumpAndSettle();
+
+      // Find the "Ignore" buttons and click the first four.
+      final buttonFinder = find.text('Ignore');
+      expect(
+        buttonFinder,
+        findsWidgets,
+      ); // Check that multiple "Ignore" buttons exist
+
+      // Tap the first four "Ignore" buttons found
+      for (int i = 0; i < 4; i++) {
+        await tester.tap(buttonFinder.at(i));
+        await tester.pumpAndSettle();
+      }
+
+      // Navigate to the "Transform" tab.
+      final transformTabFinder = find.text('Transform');
+      expect(
+        transformTabFinder,
+        findsOneWidget,
+      ); // Ensure the "Transform" tab exists.
+      await tester.tap(transformTabFinder); // Tap the "Transform" tab.
+      await tester.pumpAndSettle(); // Wait for the UI to update.
+
+      // Navigate to the "Cleanup" sub-tab within the Transform tab.
+      final cleanupSubTabFinder = find.text('Cleanup');
+      expect(
+        cleanupSubTabFinder,
+        findsOneWidget,
+      ); // Ensure the "Cleanup" sub-tab exists.
+      await tester.tap(cleanupSubTabFinder); // Tap the "Cleanup" sub-tab.
+      await tester.pumpAndSettle(); // Wait for the UI to update.
+
+      // Locate the "Ignore" chip. Adjust the Finder if needed.
+      final ignoreChipFinder =
+          find.text('Ignored'); // Adjusted finder: search by text
+      // Check if the Ignore text exists anywhere in the current widget tree.
+      expect(
+        ignoreChipFinder,
+        findsOneWidget,
+      ); // Ensure at least one "Ignore" widget exists.
+      await tester
+          .tap(ignoreChipFinder); // Tap on the single found "Ignore" chip.
+
+      await tester.tap(ignoreChipFinder); // Tap on the "Ignore" chip.
+      await tester
+          .pumpAndSettle(); // Wait for the UI to settle after the interaction.
+
+      // Tap the "Delete from Dataset" button.
+      final deleteButtonFinder = find.text('Delete from Dataset');
+      expect(
+        deleteButtonFinder,
+        findsOneWidget,
+      ); // Ensure the "Delete from Dataset" button exists.
```

Comment:
Please use the style of the comment on line by itself with space before and after all the way through the file.